config/supermicro.go

```diff
@@ -114,13 +125,208 @@ func (cm *supermicroVendorConfig) Marshal() (string, error) {
 	}
 }
 
+func (cm *supermicroVendorConfig) Unmarshal(cfgData string) (err error) {
+	// the xml exported by sum is ISO-8859-1 encoded
+	decoder := xml.NewDecoder(bytes.NewReader([]byte(cfgData)))
+	// convert characters from non-UTF-8 to UTF-8
+	decoder.CharsetReader = charset.NewReaderLabel
+
+	err = decoder.Decode(cm.ConfigData.BiosCfg)
```

Comment:
assignment to `err` can be skipped
```suggestion
	return decoder.Decode(cm.ConfigData.BiosCfg)
```